src/components/DataTable/tests/DataTable.test.tsx

```diff
@@ -139,107 +137,105 @@ describe('<DataTable />', () => {
         'Heading 2',
         'Heading 3',
       ];
-      const dataTable = mountWithAppProvider(
+      const dataTable = mountWithApp(
         <DataTable {...defaultProps} headings={headings} />,
       );
 
-      const checkbox = dataTable.find('thead tr').find(Checkbox);
+      const headerCell = dataTable.find('thead')!.find('tr')!;
 
-      expect(checkbox).toHaveLength(1);
-      expect(checkbox.prop('label')).toBe('Heading 1');
+      expect(headerCell).toContainReactComponent(Checkbox);
+      expect(headerCell.find(Checkbox)).toHaveReactProps({label: 'Heading 1'});
```

Comment:
```suggestion
      expect(headerCell).toContainReactComponent(Checkbox, {label: 'Heading 1'});
```

These could be done in a single assertion